sdk/core/core-rest-pipeline/documentation/pipeline.md

```diff
@@ -0,0 +1,251 @@
+
+# Default Pipeline
+
+`@azure/core-rest-pipeline` exposes `createPipelineFromOptions` which creates pipeline with a predefined set of policies. The resulting pipeline can be configured through `PipelineOptions`
+
+```typescript
+/**
+ * Defines options that are used to configure the HTTP pipeline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: PipelineRetryOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxySettings;
+
+  /**
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: RedirectPolicyOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options - Options to configure a custom pipeline.
+ */
+declare function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline;
+```
+
+**Note:** `InternalPipelineOptions` is used internally to configure the SDK pipeline but it is not exposed to consumers. Consumers can configure the default pipeline through the `PipelineOptions` when instantiating a new client
+
+### Policies included in a default pipeline
+
+- proxyPolicy <img src="https://img.shields.io/badge/node.js-6DA55F?style=for-the-badge&logo=node.js&logoColor=white" alt="NodeJS Only" title="NodeJS Only" width="60"/>
+  - A policy that allows one to apply proxy settings to all requests. If not passed static settings, they will be retrieved from the HTTPS_PROXY or HTTP_PROXY environment variables.
+- decompressResponsePolicy <img src="https://img.shields.io/badge/node.js-6DA55F?style=for-the-badge&logo=node.js&logoColor=white" alt="NodeJS Only" title="NodeJS Only" width="60"/>
+  - A policy to enable response decompression according to Accept-Encoding header
+- formDataPolicy
+  - A policy that encodes FormData on the request into the body.
+- userAgentPolicy
+  - A policy that sets the User-Agent header (or equivalent) to reflect the library version
+- setClientRequestIdPolicy
+  - The programmatic identifier of the setClientRequestIdPolicy
+- defaultRetryPolicy
+  - A policy that retries according to three strategies:
+    - When the server sends a 429 response with a Retry-After header.
+    - When there are errors in the underlying transport layer (e.g. DNS lookup failures).
+    - Or otherwise if the outgoing request fails, it will retry with an exponentially increasing delay.
```

Comment:
```suggestion
    - After considering the above, if the outgoing request fails with a non-final response code, it will retry with an exponentially increasing delay.
```